js_modules/dagster-ui/packages/ui-components/src/components/SplitPanelContainer.tsx

```diff
@@ -13,220 +11,185 @@ interface SplitPanelContainerProps {
-  SplitPanelContainerProps,
-  SplitPanelContainerState
-> {
-  constructor(props: SplitPanelContainerProps) {
-    super(props);
-
-    const key = `dagster.panel-width.${this.props.identifier}`;
-    const value = window.localStorage.getItem(key);
-    let size = Number(value);
-    if (value === null || isNaN(size)) {
-      size = this.props.firstInitialPercent;
-    }
-
-    this.state = {size, key, resizing: false};
-  }
+export type SplitPanelContainerHandle = {
+  getSize: () => number;
+  changeSize: (value: number) => void;
+};
 
-  onChangeSize = (size: number) => {
-    this.setState({size});
-    window.localStorage.setItem(this.state.key, `${size}`);
-  };
+export const SplitPanelContainer = forwardRef<SplitPanelContainerHandle, SplitPanelContainerProps>(
+  (props, ref) => {
+    const {
+      axis = 'horizontal',
+      identifier,
+      first,
+      firstInitialPercent,
+      firstMinSize,
+      secondMinSize = 0,
+      second,
+    } = props;
+
+    const [_, setTrigger] = useState(0);
+    const [resizing, setResizing] = useState(false);
+    const key = `dagster.panel-width.${identifier}`;
+
+    const getSize = useCallback(() => {
+      if (!second) {
+        return 100;
+      }
+      const storedSize = window.localStorage.getItem(key);
+      const parsed = storedSize === null ? null : parseFloat(storedSize);
+      return parsed === null || isNaN(parsed) ? firstInitialPercent : parsed;
+    }, [firstInitialPercent, key, second]);
+
+    const onChangeSize = useCallback(
+      (newValue: number) => {
+        window.localStorage.setItem(key, `${newValue}`);
+        setTrigger((current) => (current ? 0 : 1));
+      },
+      [key],
+    );
 
-  getSize = () => {
-    return this.state.size;
-  };
+    useImperativeHandle(ref, () => ({getSize, changeSize: onChangeSize}), [onChangeSize, getSize]);
 
-  render() {
-    const {firstMinSize, first, second} = this.props;
-    const {size: _size, resizing} = this.state;
-    const axis = this.props.axis || 'horizontal';
+    const firstSize = getSize();
 
     const firstPaneStyles: React.CSSProperties = {flexShrink: 0};
-    const firstSize = second ? _size : 100;
 
     // Note: The divider appears after the first panel, so making the first panel 100% wide
     // hides the divider offscreen. To prevent this, we subtract the divider depth.
     if (axis === 'horizontal') {
       firstPaneStyles.minWidth = firstMinSize;
-      firstPaneStyles.width = `calc(${firstSize}% - ${DIVIDER_THICKNESS}px)`;
+      firstPaneStyles.width = `calc(${firstSize / 100} * (100% - ${
+        DIVIDER_THICKNESS + (second ? secondMinSize : 0)
+      }px))`;
     } else {
       firstPaneStyles.minHeight = firstMinSize;
-      firstPaneStyles.height = `calc(${firstSize}% - ${DIVIDER_THICKNESS}px)`;
+      firstPaneStyles.height = `calc(${firstSize / 100} * (100% - ${
+        DIVIDER_THICKNESS + (second ? secondMinSize : 0)
+      }px))`;
```

Comment:
The noteworthy calculation change to accommodate minimum size on the second panel.